crates/l2/storage/src/store.rs

```diff
@@ -355,4 +355,25 @@ impl Store {
             .delete_proof_by_batch_and_type(batch_number, proof_type)
             .await
     }
+
+    pub async fn store_prover_input_by_batch_and_version(
+        &self,
+        batch_number: u64,
+        prover_version: String,
```

Comment:
I think this can be a `&str`. This way, we can save some `.clone()`s when using SQL store. In case of InMemory store, the clone will be needed anyway, but that's not used in production

Reply:
[40c2d2c](https://github.com/lambdaclass/ethrex/pull/5006/commits/40c2d2cf8f339e49de65ca716d692f1ae503490f)